libs/core/src/lib/content-density/providers/local-content-density-storage.ts

```diff
@@ -0,0 +1,44 @@
+import { Inject, Injectable } from '@angular/core';
+import { LocalStorageService } from '@fundamental-ngx/core/utils';
+import { Observable, of, Subject } from 'rxjs';
+import { ContentDensityStorage } from '../classes/abstract-content-density-storage';
+import { DEFAULT_CONTENT_DENSITY } from '../tokens/default-content-density.token';
+import { CONTENT_DENSITY_STORAGE_KEY } from '../tokens/content-density-storage-key.token';
+import { ContentDensityMode } from '../content-density.types';
+
+@Injectable()
+export class LocalContentDensityStorage implements ContentDensityStorage {
```

Comment:
actually with local storage based implementation we can act directly on the change event in the storage

Check this implementation of reactive local storage, that also tracks cross-window events
[local-storage.service.txt](https://github.com/SAP/fundamental-ngx/files/8902726/local-storage.service.txt)

Reply:
We can do that in utils separately. would be nice to have

Reply:
As you with 🤷 You're creating that local storage service anyway, so I thought it would be better to have more feature rich implementation